internal/controller/thanosquery_controller.go

```diff
@@ -132,29 +143,122 @@ func (r *ThanosQueryReconciler) buildQuerier(query monitoringthanosiov1alpha1.Th
 		LogFormat: query.Spec.LogFormat,
 	}.ApplyDefaults()
 
+	endpoints, err := r.getStoreAPIServiceEndpoints(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+
 	return manifestquery.BuildQuerier(manifestquery.QuerierOptions{
 		Options:       metaOpts,
 		ReplicaLabels: query.Spec.QuerierReplicaLabels,
 		Timeout:       "15m",
 		LookbackDelta: "5m",
 		MaxConcurrent: 20,
+		Endpoints:     endpoints,
+	}), nil
+}
 
-		// TODO(saswatamcode): Add logic to for automated tracking of endpoints.
-		Endpoints: []manifestquery.Endpoint{{
-			ServiceName: "thanos-receive",
-			Namespace:   query.GetNamespace(),
-			Type:        manifestquery.Regular,
-		}},
-	})
+// getStoreAPIServiceEndpoints returns the list of endpoints for the StoreAPI services that match the ThanosQuery storeLabelSelector.
+func (r *ThanosQueryReconciler) getStoreAPIServiceEndpoints(ctx context.Context, query monitoringthanosiov1alpha1.ThanosQuery) ([]manifestquery.Endpoint, error) {
+	selector, err := metav1.LabelSelectorAsSelector(query.Spec.StoreLabelSelector)
```

Comment:
i think you could avoid the error here by using the `client.MatchingLabels` func to return the list opts?